src/emitters/emitter.sqns.ts

```diff
@@ -92,6 +89,30 @@ export class SqnsEmitter implements IEmitter {
     await this.createEventSourceMappings();
   }
 
+  private validateAwsConfig() {
+    if (!this.options.awsConfig) {
+      throw new Error(
+        `awsConfig is required in options when using external broker.`
+      );
+    } else {
+      const { region, accountId } = this.options.awsConfig;
+      if (!accountId)
+        this.getAwsAccountId(region).then((accountId) => {
```

Comment:
use async/await and try/catch

Reply:
There is a reason not using `async await` here because we have to call its outer function in constructor? I don't know can constructor supports asynchronous calls

Reply:
Lets discuss this
A lot of functions rely on accountId and making this asynchronous would mean that those functions break if called before the accountId is set